ca/certificates.go

```diff
@@ -116,7 +115,6 @@ func (rca *RootCA) CanSign() bool {
 func (rca *RootCA) IssueAndSaveNewCertificates(paths CertPaths, cn, ou, org string) (*tls.Certificate, error) {
 	csr, key, err := GenerateAndWriteNewKey(paths)
 	if err != nil {
-		log.Debugf("error when generating new node certs: %v", err)
```

Comment:
I get that we don't want to log and then also return the error, because we'll generate confusing redundant log messages. But I think we should wrap the error messages we return, so they aren't completely out of context (for example, you don't want to see "No such file or directory" without knowing what hit this error).

So I'd change the `return nil, err` to `return nil, fmt.Errorf("error when generating new node certs: %v", err)`.

I know @stevvooe will suggest using a more sophisticated approach that doesn't lose the type information :). That's okay with me too, but I don't think it's necessary for this code right now.